### ISSUE TITLE
Inconsistent output of HeteroLinear when '"is_sorted=False"

### ISSUE THREAD

**Issue by toenshoff** [pyg-team/pytorch_geometric#6199]
### 🐛 Describe the bug

Hey,

while working with the HeteroLinear Module we observed some unexpected output behavior when using the "is_sorted=False" option.

Specifically, when pyg-lib is installed the forward pass of HeteroLinear internally sorts the vertices by type before before applying its transformation. However, the original vertex ordering is not restored before the output is returned. Instead, the updated vertex embeddings are returned in a new order sorted by type.
This seems like undesirable output behavior, since the permutation of the sorting is not returned and can also not be safely recovered by the user as the used sorting method is not stable.

Importantly, this change in vertex order does not happen when pyg-lib is not installed, since then no sorting is used in the forward pass. Therefore, this behavior is inconsistent between different installations.

Here is a minimal example to reproduce this:

```python
import torch
from torch_geometric.nn import HeteroLinear

lin = HeteroLinear(1, 1, num_types=2, bias=False, is_sorted=False)

x = torch.tensor([0.0, 1.0, 0.0]).view(-1, 1)
type_vec = torch.tensor([0, 1, 0])

y = lin(x, type_vec)
print(y.detach().flatten().numpy())
```
With pyg-lib installed we obtain:
```
[ 0.          0.         -0.76703763]
```

Without pyg-lib installed we obtain:
```
[0.         0.58155024 0.        ]
```

Note that we did not fix a seed here, but crucially the order of the output is different in each case.

If this behavior is intended, it should be clearly specified in the documentation.
Otherwise, we have created a pull request (https://github.com/pyg-team/pytorch_geometric/pull/6198) which returns nodes to their original order when pyg-lib is installed and "is_sorted=False": 

Kind Regards,
Jan

### Environment

* PyG version: 2.2.0
* PyTorch version: 1.13.0
* OS: Ubuntu 22.04 LTS
* Python version: 3.10
* CUDA/cuDNN version: 11.7
* How you installed PyTorch and PyG (`conda`, `pip`, source): conda
* Any other relevant information (*e.g.*, version of `torch-scatter`):  pyg-lib-0.1.0 was installed
